rpc/codec/conn.go

```diff
@@ -5,92 +5,85 @@
 package codec
 
 import (
+	"bufio"
 	"encoding/binary"
-	"errors"
 	"io"
 	"net"
+	"sync"
+
+	"github.com/gogo/protobuf/proto"
 )
 
+type frameBuf struct {
+	size [binary.MaxVarintLen64]byte
+}
+
+var framePool = sync.Pool{
+	New: func() interface{} {
+		return &frameBuf{}
+	},
+}
+
 func sendFrame(w io.Writer, data []byte) (err error) {
 	// Allocate enough space for the biggest uvarint
-	var size [binary.MaxVarintLen64]byte
+	buf := framePool.Get().(*frameBuf)
```

Comment:
Why this change? Did the escape analysis decide that this array couldn't be put on the stack?